File: src/remote-helper.ts
import { AppConfig } from "./config.js";
import { remotePathExpr, runSsh, shellQuote, CommandResult } from "./ssh.js";

export async function runRemoteHelper(config: AppConfig, args: string[], input?: string | Buffer): Promise<CommandResult> {
  const command = [remotePathExpr(config.remoteHelperPath), ...args.map(shellQuote)].join(" ");
  return runSsh(config.host, command, input);
}

File: src-rs/remote_helper.rs
use anyhow::Result;

use crate::config::AppConfig;
use crate::ssh::{CommandResult, remote_path_expr, run_ssh, shell_quote};

pub fn run_remote_helper(
    config: &AppConfig,
    args: &[String],
    input: Option<&[u8]>,
) -> Result<CommandResult> {
    let command = std::iter::once(remote_path_expr(&config.remote_helper_path))
        .chain(args.iter().map(|arg| shell_quote(arg)))
        .collect::<Vec<_>>()
        .join(" ");
    run_ssh(config, &command, input)
}

